config/types/etcd.go

```diff
@@ -373,6 +385,178 @@ type Etcd3_3 struct {
 	ExperimentalEnableV2V3          *string `yaml:"experimental_enable_v2v3"           cli:"experimental-enable-v2v3"`
 }
 
+type Etcd3_4 struct {
+	Name                                    *string `yaml:"name"                                        cli:"name"`
+	DataDir                                 *string `yaml:"data_dir"                                    cli:"data-dir"`
```

Comment:
Maybe types could be embedded to deduplicate most of the fields?

Reply:
No I don't think that makes sense. I've compared 3.3 to 3.4 and 3.5, every version adds and removes options, and trying to separate that into something that can be embedded will require more maintenance than parsing the options for a specific release and generating a struct from that.

Reply:
Sure, sounds good :+1: I wonder if we could have the code you wrote for generation in `go:generate` or something.

Reply:
I'll look at that next time we need to update, right now it would be cumbersome to do so as it requires pulling all etcd binaries.